### ISSUE TITLE
empty_template() option broken

### ISSUE THREAD

**Issue by tversteeg** [museun/markings#11]
Here's a small test to show that the `empty_template()` option doesn't work as intended:

```rust
fn empty_template_bug() {
    let template =
        Template::parse("${short_name}", Opts::default().empty_template().build()).unwrap();
    let parts = Args::new().with("short_name", &1).build();
    assert_eq!("1", template.apply(&parts).unwrap());
}
```

As you can see `${short_name}` isn't replaced with `1` in this example.

Thanks for your work on this library!

**Comment by museun** [comment]
I don't know what it should do here, `empty_template` simple tells it that the template shouldn't be replaced (e.g. there might not be a template).

Should `parse` return an error if it does find a template yet you've specified its empty?

I think perhaps the name should be changed, `non_applicable` or similar

**Comment by tversteeg** [comment]
Ah right, I assumed `empty_template` allowed a file that didn't contain any of the arguments. How can I allow this if `empty_template` doesn't do that?

**Comment by museun** [comment]
I see, I can adjust the apply method to have that behavior where it won't produce an error if you've marked it as possibly empty. I was looking at it from the opposite point of view, but that makes sense now.

**Comment by museun** [comment]
I've changed it in #12 but I would need some examples to make sure this is the correct behavior.

You can use 
```toml
markings = { git = "https://github.com/museun/markings", branch = "empty" }
```

To test the git branch

**Comment by tversteeg** [comment]
Thanks a lot! I won't be able to test it today so I'll let you know tomorrow.

**Comment by tversteeg** [comment]
Yes this seems to work, thanks a lot!

**Comment by museun** [comment]
I've published it as 0.3.2. Thanks for the assistance.
